gammapy/image/maps.py

```diff
@@ -341,12 +341,13 @@ def reproject(self, refheader, mode='interp', *args, **kwargs):
         skymap : `SkyMap`
             Skymap reprojected onto ``refheader``.
         """
+
         from reproject import reproject_interp
         out = reproject_interp(
             input_data=self.to_image_hdu(),
             output_projection=refheader,
             *args,
-            **kwargs,
+            **kwargs
         )
         map = SkyMap(name=self.name, data=out[0], wcs=self.wcs, unit=self.unit, meta=self.meta)
```

Comment:
Currently the wrong `WCS` is passed to `SkyMap`, it should be created from the new reference header. I would slightly prefer if `reproject` works "in place" i.e. changes the current skymap instead of returning a changed and implicit copy of it. We can easily add a `.copy()` method to the sky map class (which might be a good idea anyway...).